Doc/whatsnew/3.10.rst

```diff
@@ -124,6 +124,163 @@ Check :pep:`617` for more details.
 (Contributed by Guido van Rossum, Pablo Galindo and Lysandros Nikolaou
 in :issue:`12782` and :issue:`40334`.)
 
+PEP 634: Structural Pattern Matching
+------------------------------------
+
+Pattern matching for objects (sequences, mappings, primitive data types
+as well as class instances) has been added in the form of a *match
+statement*.  Here's a quick tutorial (more docs are forthcoming):
+
+A match statement takes an expression and compares its value to successive
+patterns given as one or more case blocks.  This is superficially
+similar to a switch statement in C, Java or JavaScript (and many
+other languages), but much more powerful.
+
+The simplest form compares a subject value against one or more literals::
+
+    def http_error(status):
+        match status:
+            case 400:
+                return "Bad request"
+            case 404:
+                return "Not found"
+            case 418:
+                return "I'm a teapot"
+            case _:
+                return "Something's wrong with the Internet"
+
+Note the last block: the "variable name" ``_`` acts as a *wildcard* and
+never fails to match.
+
+You can combine several literals in a single pattern using ``|`` ("or")::
+
+            case 401 | 403 | 404:
+                return "Not allowed"
```

Comment:
This example has some problems, as mentioned in https://bugs.python.org/issue42128#msg387371:

> I've seen too many knee-jerk reactions over the past weeks along the lines of "the new switch feature can't handle named constants!".

Interestingly I haven't seen such reactions, I've just seen a lot of (mostly) positive responses to my tweet linking to this same tutorial: https://twitter.com/gvanrossum/status/1361124478671933443. (Admittedly, there were negative reactions too, including "the worst pep of the python history." But I didn't see anything about named constants.)